samples/drivers/charger/src/main.c

```diff
@@ -32,6 +32,29 @@ int main(void)
 
 	printk("Found device \"%s\", getting charger data\n", chgdev->name);
 
+#ifdef CONFIG_CHARGER_DISCHARGE_CURRENT_NOTIFICATIONS
+	val.discharge_current_notification.current_ua =
+		CONFIG_APP_DISCHARGE_CURRENT_NOTIFICATION_THRESHOLD_UA;
+	val.discharge_current_notification.severity = CHARGER_SEVERITY_WARNING;
+	val.discharge_current_notification.duration_us =
+		CONFIG_APP_DISCHARGE_CURRENT_NOTIFICATION_DURATION_US;
```

Comment:
Hey does this really have to be part of the runtime API? At first looks it seems like a device specific configuration that should just be setup for the specific device at init time, the parameters be in the binding and never touched again.

Reply:
It is necessary for ECs managing charging/discharging on x86 platforms. If this were an absolute maximum trip point that protected the battery pack, I would agree that yes it could be set and forgotten at init. 

However the purpose of these notification events is not to protect the battery but rather to prevent a system brownout by having the charger notify the system of excess power being drawn from the battery. The threshold at which the charger should notify the system can vary depending on the known SOH/SOC of the battery.

Examples of conditions that could reduce maximum current you could draw from the battery aside from SOC:
- Heat
- Cycle count
- Cold
- The absence of an external power source
- The time between turbo event (load frequency)

This feature is supported on the TI, Intersil, and Southchip chargers targeting x86 platforms. I could see a case for using a Kconfig to enable/disable the `CHARGER_PROP_*_NOTIFICATION` props.